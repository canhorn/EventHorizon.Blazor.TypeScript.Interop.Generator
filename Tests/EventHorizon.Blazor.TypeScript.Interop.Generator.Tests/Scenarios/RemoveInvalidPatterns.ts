declare module Examples {
  export class Observable<T> {}
  
  export class ExampleClass {
    onBeforeTextureInitObservable: Observable<
      import("babylonjs/Materials/Textures/texture").Texture
    >;
  }
}
