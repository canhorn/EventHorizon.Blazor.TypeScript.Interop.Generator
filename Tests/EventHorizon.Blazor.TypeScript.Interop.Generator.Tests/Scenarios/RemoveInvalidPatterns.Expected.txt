/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties
        private Observable<CachedEntityObject> __onBeforeTextureInitObservable;
        public Observable<CachedEntityObject> onBeforeTextureInitObservable
        {
            get
            {
            if(__onBeforeTextureInitObservable == null)
            {
                __onBeforeTextureInitObservable = EventHorizonBlazorInterop.GetClass<Observable<CachedEntityObject>>(
                    this.___guid,
                    "onBeforeTextureInitObservable",
                    (entity) =>
                    {
                        return new Observable<CachedEntityObject>() { ___guid = entity.___guid };
                    }
                );
            }
            return __onBeforeTextureInitObservable;
            }
            set
            {
__onBeforeTextureInitObservable = null;
                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "onBeforeTextureInitObservable",
                    value
                );
            }
        }
        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}